src/_isCyclic.ts

```diff
@@ -0,0 +1,53 @@
+const MAX_DEPTH = 1000;
```

Comment:
By the way, why is this function / this limit needed?

Given the limit, this function is not "fair". If the limit is exceeded, the function will say that the object is not cyclic, although it can be cyclic. Why not remove the limit?

If we are worried that the cycle check might take a long time because the check is called in a dipmerge, there is a simple solution - do not call the cycle check. Instead of pre-checking, add a runtime check that will check nodes as they are copied, and will respond to an attempt to process circular references. Then the check will become part of the implementation and will not be needed as a separate function.

Reply:
Removed this